tests/fsharpqa/Source/Conformance/SpecialAttributesAndTypes/Imported/CallerInfo/CallerMemberName.fs

```diff
@@ -0,0 +1,150 @@
+namespace Test
+
+open System.Runtime.CompilerServices
+open System.Reflection
+open CSharpLib
+
+[<assembly:MyCallerMemberNameAttribute>]
+do
+    ()
+
+[<MyCallerMemberName()>]
+type MyTy() =
+    let functionVal = MyTy.GetCallerMemberName
+    let typeLetValue = MyTy.GetCallerMemberName()
+    let typeLetFunc (i:int) = i, MyTy.GetCallerMemberName()
+    let typeLetFuncNested () =
```

Comment:
Could you please also specify and test what happens when the call to GetCallerMemberName is made from 

* an anonymous lambda experession, e.g. ``(fun () -> ...)``.  
* an object expression member implementation implementing an interface
* an object expression member implementation implementing an abstract member in a base class
* a delegate implementation e.g. ``new System.Func<int,int>(fun a -> ...)``
* within an async expression ``async { ... }`` (whose desugaring has some implied lambda expressions)

Thanks